Apps/Common/src/ErrorHandling/ProblemDetailsExtensions.cs

```diff
@@ -0,0 +1,117 @@
+﻿// -------------------------------------------------------------------------
+//  Copyright © 2019 Province of British Columbia
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace HealthGateway.Common.ErrorHandling
+{
+    using System;
+    using System.ServiceModel;
+    using FluentValidation;
+    using HealthGateway.Common.ErrorHandling.Exceptions;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+
+    /// <summary>
+    /// Extension methods for <see cref="Exception"/>.
+    /// Used to transform exceptions into problem details responses.
+    /// </summary>
+    public static class ProblemDetailsExtensions
+    {
+        /// <summary>
+        /// Transforms an exception into a problem details response.
+        /// </summary>
+        /// <param name="exception">An <see cref="Exception"/>.</param>
+        /// <param name="httpContext">The HTTP context of the request.</param>
+        /// <param name="includeException">Used to determine if the exception details are passed to the client.</param>
+        /// <returns>A <see cref="ProblemDetails"/> model to return the consumer.</returns>
+        public static ProblemDetails ToProblemDetails(this Exception exception, HttpContext httpContext, bool includeException = false)
+        {
+            return TransformException(exception, httpContext, includeException);
+        }
```

Comment:
Good point, I will consider moving the type checks into the extension method to reliably determine the underlying type. This is a design in progress so it could change. I see the need to check and cast to the appropriate type for every `ToProblemDetails` call is error prone and leaves the responsibility with the developer to remember to cast before calling.

Good catch. I'm going to handle this in AB#16608 as this really relates to the handler side of the implementation.

Reply:
creating a list of these comments to address check AB#16608 for a full list of items to address.